### DISCUSSION TITLE
Help Needed with FastAPI Dynamic Enum Integration Error

### DISCUSSION THREAD

**Discussion by VaishnaviPandey02** [fastapi/fastapi#12087] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum
from typing import List
from sqlalchemy.orm import Session
 
# Assuming these functions are defined to get data from the database
def get_customers(db: Session) -> List[str]:
    customers = db.query(CoreCustomer.customer_name).all()
    return [customer[0] for customer in customers]
 
def get_providers(db: Session) -> List[str]:
    providers = db.query(WasteProvider.provider_name).all()
    return [provider[0] for provider in providers]
 
def get_waste_categories(db: Session) -> List[str]:
    waste_categories = db.query(WasteCategory.category_name).all()
    return [category[0] for category in waste_categories]
 
# Function to create dynamic Enums
def create_enum(enum_name: str, enum_values: List[str]):
    return Enum(enum_name, {value: value for value in enum_values})
 
# Example usage with database session
def setup_dynamic_enums(db: Session):
    global CustomerEnum, ProviderEnum, WasteCategoryEnum
 
    # Fetch values from database
    customer_names = get_customers(db)
    provider_names = get_providers(db)
    waste_category_names = get_waste_categories(db)
 
    # Create dynamic Enums
    CustomerEnum = create_enum('CustomerEnum', customer_names)
    ProviderEnum = create_enum('ProviderEnum', provider_names)
    WasteCategoryEnum = create_enum('WasteCategoryEnum', waste_category_names)

@router.post("/my_endpoint/")
async def my_endpoint(
        customer: CustomerEnum = Form(...),
        provider: ProviderEnum = Form(...),
        waste_category: WasteCategoryEnum = Form(...),
        pickup_date_column: str = Form(...),
        facility_column: str = Form(...),
        waste_name_column: str = Form(...),
        weight_column: str = Form(...),
        is_recycled_column: str = Form(...),
        is_diverted_column: str = Form(...),
        unit_column: str = Form(...),
        file: UploadFile = File(...),
        db: Session = Depends(get_db)
):
          pass

in main.py

if __name__ == "__main__":
     db_session = get_db()
     setup_dynamic_enums(db_session)
```

### Description

How can I dynamically populate a dropdown in a Swagger UI based on a dynamic enum? I am able to achieve this using a static enum.

### Operating System

Linux

### FastAPI Version

fastapi==0.111.1

### Pydantic Version

pydantic==2.8.2

### Python Version

python==3.10

**Comment by YuriiMotov** [comment, marked as answer]
You can manually modify openapi schema every time you need to update enum items. More about openapi schema modifying: https://fastapi.tiangolo.com/how-to/extending-openapi/#generate-the-openapi-schema

But I don't think it's a good idea. The documentation of your app shouldn't be changed during the execution of app.

I would better create an endpoint that returns list of possible options. Client will be able to request list of options any time and use one of them. Although it will not let you choose this options from drop-down menu in Swagger..
